TerriaJS/terriajs
# Strongly type `item` in WorkbenchItem and remove `show` toggle for non `Mappable` items

## Strongly type `item` in WorkbenchItem and remove `show` toggle for non `Mappable` items
  
### Test me

- http://ci.terria.io/workbench-type-item/
- "Explore map data"
- Add "Magda Item" to the workbench
- Notice that item doesn't have the "show" toggle box while it is loading
  - This is because the `MagdaReference` isn't mappable - until it has loaded the reference
- **Note** WMS item will throw errors

### Checklist
